packages/docusaurus-theme-common/src/hooks/useCodeWordWrap.ts

```diff
@@ -17,6 +17,8 @@ export function useCodeWordWrap(): {
   const [isEnabled, setIsEnabled] = useState(false);
   const [isCodeScrollable, setIsCodeScrollable] = useState<boolean>(false);
   const codeBlockRef = useRef<HTMLPreElement>(null);
+  const [mutationObserver, setMutationObserver] =
+    useState<MutationObserver | null>(null);
```

Comment:
Should this be a state or a ref? We have usage of `IntersectionObserver` in core, and we use a ref there.

Reply:
We shouldn't need to re-render after IO is set, so ref is better 👍